tests/channels/test_nbody_reactions.py

```diff
@@ -27,164 +25,138 @@ def reduce_violated_rules(
-
-    graph_interaction_settings = stm.prepare_graphs()
-    result = stm.find_solutions(graph_interaction_settings)
-
-    reduced_violations = reduce_violated_rules(result.violated_node_rules)
-    assert reduced_violations == set(expected)
-
-
-@pytest.mark.parametrize(
-    "test_input,expected",
-    [
-        ((["f(0)(980)"], ["pi+", "pi-"]), []),
-        # (
-        #     (["J/psi(1S)"], ["pi0", "f(0)(980)"]),
-        #     ["IsoSpinConservation", "CParityConservation", "SpinConservation"],
-        # ),
-        ((["p", "p"], ["Sigma+", "n", "K0", "pi+", "pi0"]), []),
         (
-            (["p", "p"], ["Sigma+", "n", "K~0", "pi+", "pi0"]),
-            ["StrangenessConservation", "isospin_conservation"],
+            (["Xi0"], ["p", "pi-"]),
+            {"StrangenessConservation", "isospin_conservation"},
+        ),
+        ((["pi-", "p"], ["Lambda", "K0"]), {}),
+        ((["pi0"], ["gamma", "gamma"]), {}),
+        ((["pi0"], ["gamma", "gamma", "gamma"]), {"c_parity_conservation"}),
+        ((["Sigma-"], ["n", "e-", "nu(e)~"]), {"StrangenessConservation"}),
+        (
+            (["rho(770)0"], ["pi0", "pi0"]),
+            {
+                "isospin_conservation",  # Clebsch Gordan coefficient = 0
+                "c_parity_conservation",
+                "identical_particle_symmetrization",
+            },
+        ),
+        ((["rho(770)0"], ["gamma", "gamma"]), {"c_parity_conservation"}),
+        ((["rho(770)0"], ["gamma", "gamma", "gamma"]), {}),
+        (
+            (["J/psi(1S)"], ["pi0", "eta"]),
+            {"c_parity_conservation", "isospin_conservation"},
+        ),
+        (
+            (["J/psi(1S)"], ["rho(770)0", "rho(770)0"]),
+            {"c_parity_conservation", "g_parity_conservation"},
+        ),
+        (
+            (["K~0"], ["pi+", "pi-", "pi0"]),
+            {"isospin_conservation", "StrangenessConservation"},
         ),
     ],
 )
-def test_strong_reactions(test_input, expected):
-    stm = StateTransitionManager(
+def test_nbody_reaction(test_input, expected):
+    # define all of the different decay scenarios
+    print("processing case:" + str(test_input))
+
+    violations = check_reaction_violations(
         initial_state=test_input[0],
         final_state=test_input[1],
-        formalism_type="canonical",
-        topology_building="nbody"
-        if len(test_input[0]) > 1 or len(test_input[1]) > 2
-        else "isobar",
-        solving_mode=SolvingMode.Full,
-        number_of_threads=1,
     )
 
-    stm.set_allowed_interaction_types([InteractionTypes.Strong])
-
-    graph_interaction_settings = stm.prepare_graphs()
-    result = stm.find_solutions(graph_interaction_settings)
-
-    reduced_violations = reduce_violated_rules(result.violated_node_rules)
-    assert set(reduced_violations) == set(expected)
+    reduced_violations = reduce_violated_rules(violations)
+    print("not allowed! violates: " + str(reduced_violations))
```

Comment:
I would just remove all print statements in tests (and replace them with `assert`s where possible). Same with line 153

Reply:
44e8ea04bba93a7a35021d254c48a9afa1d045b3